packages/util-dns/src/NodeDnsResolveHostResolver.ts

```diff
@@ -0,0 +1,51 @@
+import { HostAddress, HostAddressType, HostResolver as IHostResolver, HostResolverArguments } from "@aws-sdk/types";
+import { promises as dnsPromises } from "dns";
+
+/**
+ * @internal
+ * {@link HostResolver} implementation that uses the Node.js dns.resolve*() APIs.
+ *
+ * TODO(dns): should NOT be used or exported until fixed to fit the specified
+ * behavior:
+ * - implement cache
+ * - happy eyeballs support
+ * - identity behavior for a valid input IP address
+ */
+export class NodeDnsResolveHostResolver implements IHostResolver {
```

Comment:
This class is here as a reference in case we want to move to the `dns.resolve*()` APIs in the future.

This is NOT exported in `util-dns` in this PR.